### PR TITLE
Rewrite the nextjs example to use Yoga

### LINKED ISSUE #761: Change the Next.js example to use Yoga?
Helix seems to be on track for deprecation: https://github.com/dotansimha/graphql-yoga/issues/1351, https://github.com/contra/graphql-helix/issues/251.

I used the Next.js example to start my own project, and only later found out about the Yoga vs Helix situation. Should I make a PR migrating the Next.js example to Yoga?

### PR DESCRIPTION
**Pull request by neongreen**
Fixes #761.

Sorry for the pnpm-lock.yaml noise, I used the same pnpm version as in `package.yaml` but it didn't help. Perhaps you can reset it and push the regenerated lockfile to my branch (assuming that it even matters).

### CODE REVIEW

**Comment by hayes** [comment]
Lock file doesn't matter as long as tests pass, Pothos doesn't have any dependencies in published packages, so shouldn't be an issue

File: examples/nextjs/pages/api/graphql.ts [hunk 1]
```diff
@@ -1,34 +1,20 @@
...
+};
 
-    void sendResult(result, res);
-  }
-}) as NextApiHandler;
+export default createYoga<{
+  req: NextApiRequest;
+  res: NextApiResponse;
+}>({
+  schema,
```
Comment (hayes) [inline examples/nextjs/pages/api/graphql.ts:18]: I know not all existing examples have this, but all examples should have a context factory that at a minimum creates an empty context object.  Some Pothos plugins require a context object, and it's good practice to set this up for any new graphql api

**Comment by hayes** [comment]
Thanks for adding this!

File: examples/nextjs/pages/api/graphql.ts (hunk 1 above)
Comment (neongreen) [reply examples/nextjs/pages/api/graphql.ts:18]: I agree and I'll add a context factory

**Review by hayes — APPROVED** [review]

**Pull request merged**